flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/Table.java

```diff
@@ -83,6 +84,11 @@
 	 */
 	void printSchema();
 
+	/**
+	 * Returns underlying logical representation of this table.
+	 */
+	TableOperation getTableOperation();
```

Comment:
I agree this method has no much of use for users, though I think we should be able to work against the interface. We should not be forced to cast the `Table` in `TableEnvironment` to extract the underlying plan for the `Table`. Otherwise it does not make sense to have an interface at all.

Reply:
This method is definitely worth discussing because it fits better to Impl. On the otherhand this will be the output of the API and we don't need a lot of casting at different positions. We will mark this method as internal and can still remove it from the interface in the future.

Reply:
+1 to mark this method as internal.